### DISCUSSION TITLE
Absolute URL error

### DISCUSSION THREAD

**Discussion by dandesdev** [vercel/next.js#23697] [Help] [answered]
Hi! Im currently working on a project (https://github.com/danielx-art/linhadotempo) and Im not able to deploy that to vercel, everytime I get this error in my logs:

Unhandled error during request: TypeError: Only absolute URLs are supported
--
10:48:45.361 | at getNodeRequestOptions (/vercel/workpath0/node_modules/node-fetch/lib/index.js:1305:9)
10:48:45.361 | at /vercel/workpath0/node_modules/node-fetch/lib/index.js:1410:19
10:48:45.361 | at new Promise (<anonymous>)
10:48:45.362 | at fetch (/vercel/workpath0/node_modules/node-fetch/lib/index.js:1407:9)
10:48:45.362 | at getStaticProps (/vercel/workpath0/.next/serverless/pages/index.js:686:22)

Here is the part of the code that seems to be generating this issue:

```
const envPath = process.env.NODE_ENV === "development" ? 
"http://localhost:3000"
: process.env.VERCEL_URL;

const bookPath = "/api/book";

export async function getStaticProps(context) {

    console.log(`The path is ${envPath + bookPath}`); //<-- This is outputting the absolute URL correctly

    const book = await fetch(envPath + bookPath).then(res => res.json());  //< --but this fetch is complaining Im not using absolute URL

    return {
        props: {
            book
        }
    }
}
```

What is going wrong? Im in my 15th attempt slightly changing things to try to find the issue but vercel keeps logging this error.
I would greatly appreciate the help! ;)

**Comment by akellbl4** [comment, marked as answer]
Hi!
Hmm, looks like you are trying to call an API route that works as a Next.js API route.
The problem might be that your next.js app hasn't started and you don't have access to it.

**Reply by dandesdev** [reply]
Hi, thanks! I solved the issue in kind of a weird way.
In the project setting in vercel website, instead of creating a system environment variable with the name 'VERCEL_URL' with the value 'VERCEL_URL', I just entered the value and vercel automatically filled the name with 'NEXT_PUBLIC_VERCEL_URL'.
Then, I wrote 
```
const envPath = process.env.NODE_ENV === "development" ? 
"http://localhost:3000"
:`https://${process.env.NEXT_PUBLIC_VERCEL_URL}`;
```

with the "https://" in front of it, instead of 

```
const envPath = process.env.NODE_ENV === "development" ? 
"http://localhost:3000"
: process.env.NEXT_PUBLIC_VERCEL_URL;
```

and it worked. However, another problem came up wich is that my fetch is returning a html response and not a json response in vercel (in local is working fine). But Im starting another topic about that.

Thank you very much!

**Reply by akellbl4** [reply]
Oh, yep make sense. They create  `NEXT_PUBLIC_VERCEL_URL` because it should be accessible on the client side.
